src/styles/blocks/button.scss

```diff
@@ -0,0 +1,78 @@
+@use '../utils/vars';
+@use '../utils/mixins' as *;
+
+.button {
+  &--type1 {
```

Comment:
Tutaj jedynie moze mogłabys uzyć lepszego nazwenictwa. Type1 mało mówi.
Patrz jak to rozwiązali np w bibliotece komponentów MUI
`<Button variant="text">Text</Button>
<Button variant="contained">Contained</Button>
<Button variant="outlined">Outlined</Button>`
https://mui.com/material-ui/react-button/